### PR TITLE
Improve Settings dialog design.

### PR DESCRIPTION
**Pull request by allexzander**
For #5142 
### Before:
https://github.com/nextcloud/desktop/assets/12224014/d0700c19-8a73-4320-915e-2d50a8e47043
### After:
https://github.com/nextcloud/desktop/assets/12224014/0c258384-e7a1-404c-99dd-b219a145cff4

### CODE REVIEW

**Review by nimishavijay — CHANGES_REQUESTED** [review]
Great improvement! Looks really nice :) 
Some small wording changes:
- [ ] "No Proxy" --> "No proxy"
- [ ] "Specify proxy manually as" --> "Manually specify proxy"

No other comments from me :)

**Comment by allexzander** [comment]
> Great improvement! Looks really nice :) Some small wording changes:
> 
> * [ ]  "No Proxy" --> "No proxy"
> * [ ]  "Specify proxy manually as" --> "Manually specify proxy"
> 
> No other comments from me :)

Done https://github.com/nextcloud/desktop/pull/6150/commits/d58dd8d2027d0003259358aee309538b035ddc80

**Review by nimishavijay — APPROVED** [review]

File: src/gui/generalsettings.cpp [hunk 1]
```diff
@@ -517,15 +519,15 @@ void GeneralSettings::slotStyleChanged()
 void GeneralSettings::customizeStyle()
 {
     // setup about section
-    QString about = Theme::instance()->about();
-    Theme::replaceLinkColorStringBackgroundAware(about);
-    _ui->aboutLabel->setText(about);
+    auto aboutText = Theme::instance()->about();
```
Comment (mgallien) [inline src/gui/generalsettings.cpp:522]:
```suggestion
    const auto aboutText = Theme::instance()->about();
```

File: src/libsync/theme.cpp [hunk 2]
```diff
@@ -543,25 +543,35 @@ QString Theme::gitSHA1() const
...
     QString devString;
     //: Example text: "<p>Nextcloud Desktop Client</p>"   (%1 is the application name)
-    devString = tr("<p>%1 Desktop Client</p>")
-              .arg(APPLICATION_NAME);
+    devString = tr("<p>%1 Desktop Client</p>").arg(APPLICATION_NAME);
 
     devString += tr("<p>Version %1. For more information please click <a href='%2'>here</a>.</p>")
-              .arg(QString::fromLatin1(MIRALL_STRINGIFY(MIRALL_VERSION)) + QString(" (%1)").arg(osName))
-              .arg(helpUrl());
+                     .arg(QString::fromLatin1(MIRALL_STRINGIFY(MIRALL_VERSION)) + QString(" (%1)").arg(osName))
```
Comment (mgallien) [inline src/libsync/theme.cpp:557]: please use the `arg` overload taking multiple strings at once. they are faster

Comment (mgallien) [reply src/libsync/theme.cpp:557]: maybe even try to change this long complicated method to build teh string by using placeholders and a single `arg` call ?

File: src/libsync/theme.cpp [hunk 3]
```diff
@@ -543,25 +543,35 @@ QString Theme::gitSHA1() const
...
+    QStringList osStringList = Utility::platformName().split(QLatin1Char(' '));
+    QString osName = osStringList.at(0);
 
-    devString += tr("<p><small>Using virtual files plugin: %1</small></p>")
-                     .arg(Vfs::modeToString(bestAvailableVfsMode()));
-    devString += QStringLiteral("<br>%1")
-              .arg(QSysInfo::productType() % QLatin1Char('-') % QSysInfo::kernelVersion());
+    QString devString;
+    //: Example text: "<p>Nextcloud Desktop Client</p>"   (%1 is the application name)
+    devString = tr("<p>%1 desktop client %2</p>").arg(APPLICATION_NAME).arg(QString::fromLatin1(MIRALL_STRINGIFY(MIRALL_VERSION)));
```
Comment (mgallien) [inline src/libsync/theme.cpp:574]: same comment about using the `arg` overload with multiple arguments

**Review by mgallien — CHANGES_REQUESTED** [review]

File: src/gui/generalsettings.cpp (hunk 1 above)
Comment (allexzander) [reply src/gui/generalsettings.cpp]: @mgallien Impossible, see` void Theme::replaceLinkColorStringBackgroundAware(QString &linkString);` that expects non-const reference

File: src/libsync/theme.cpp (hunk 2 above)
Comment (allexzander) [reply src/libsync/theme.cpp]: @mgallien modified where string translations don't break

**Review by claucambra — APPROVED** [review]

**Comment by jancborchardt** [comment]
Btw @allexzander the general style which still looks a bit outdated (grey background and the containers) I assume is blocked by having a new Qt version which looks more like Windows 10 or hopefully even 11?

**Comment by jospoortvliet** [comment]
@jancborchardt I guess tracked here https://github.com/nextcloud/desktop/pull/4584

**Pull request merged**